### ISSUE TITLE
TypeError when running examples Si/GaAs.

### ISSUE THREAD

**Issue by sayeneBB** [hackingmaterials/amset#891]
Dear Developers,

I have installed the newest version of amset, but I met errors when running Si and GaAs examples. I have listed the error messages of Si. Could you please tell me how to fix this problem? Thank you!

```~~~~~~~~~~~~~~~~~~~~~~~~~~ INTERPOLATION ~~~~~~~~~~~~~~~~~~~~~~~~~~

  Getting band interpolation coefficients
    - time: 6.6004 s

  Interpolation parameters:
    - k-point mesh: 99x99x99
    - energy cutoff: 1.5 eV

  Interpolating spin-up bands 7-10
    - time: 2.3792 s

  bandgap set to 1.330 eV, applying scissor of 1.140 eV

  Generating tetrahedron mesh vertices

  
    ERROR: amset exiting on 25 Jun 2024 at 17:41
  Traceback (most recent call last):
    File "/home/chengnp/.local/bin/amset", line 33, in <module>
      sys.exit(load_entry_point('amset', 'console_scripts', 
  'amset')())
    File "/usr/lib/python3/dist-packages/click/core.py", line 1128, 
  in __call__
      return self.main(*args, **kwargs)
    File "/usr/lib/python3/dist-packages/click/core.py", line 1053, 
  in main
      rv = self.invoke(ctx)
    File "/usr/lib/python3/dist-packages/click/core.py", line 1659, 
  in invoke
      return _process_result(sub_ctx.command.invoke(sub_ctx))
    File "/usr/lib/python3/dist-packages/click/core.py", line 1395, 
  in invoke
      return ctx.invoke(self.callback, **ctx.params)
    File "/usr/lib/python3/dist-packages/click/core.py", line 754, in
   invoke
      return __callback(*args, **kwargs)
    File "/home/chengnp/desktop/amset/amset/tools/run.py", line 139, in 
  run
      runner.run()
    File "/home/chengnp/desktop/amset/amset/core/run.py", line 65, in run
      mem_usage, (amset_data, usage_stats) = memory_usage(
    File "/home/chengnp/.local/lib/python3.10/site-
  packages/memory_profiler.py", line 379, in memory_usage
      returned = f(*args, **kw)
    File "/home/chengnp/desktop/amset/amset/core/run.py", line 114, in 
  _run_wrapper
      amset_data, interpolation_time = self._do_interpolation()
    File "/home/chengnp/desktop/amset/amset/core/run.py", line 204, in 
  _do_interpolation
      amset_data = interpolater.get_amset_data(
    File 
  "/home/chengnp/desktop/amset/amset/interpolation/bandstructure.py", line
   246, in get_amset_data
      ) = get_kpoints_tetrahedral(
    File 
  "/home/chengnp/desktop/amset/amset/electronic_structure/kpoints.py", 
  line 122, in get_kpoints_tetrahedral
      grid_mapping, grid_address = spglib.get_ir_reciprocal_mesh(
    File "/home/chengnp/.local/lib/python3.10/site-
  packages/spglib/spglib.py", line 1569, in get_ir_reciprocal_mesh
      lattice, positions, numbers, _ = _expand_cell(cell)
    File "/home/chengnp/.local/lib/python3.10/site-
  packages/spglib/spglib.py", line 1999, in _expand_cell
      lattice = np.array(np.transpose(cell[0]), dtype="double", 
  order="C")
  TypeError: float() argument must be a string or a real number, not 
  'Atom'

**Comment by sayeneBB** [comment]
```hello?```
i have figured out what is wrong. it is a version problem of my numpy.
Is this code still being mainteining?

**Comment by whduddnj** [comment]
Hello, I have met a same issue. Can you comment what version of numpy are you using?

**Comment by whduddnj** [comment]
For my case, this issue can be resolved by
https://matsci.org/t/typeerror-float-argument-must-be-a-string-or-a-real-number-not-atom/55047/3
Downgrading spglib from latest version to 2.0.0 works

**Comment by utf** [comment]
This has been fixed in v0.4.21

**Comment by sayeneBB** [comment]
> R

version of 1.25.2
